pkg/liquidity-mining/test/AuthorizerAdaptorEntrypoint.test.ts

```diff
@@ -0,0 +1,115 @@
+  });
+
+  describe('constructor', () => {
+    it('sets the vault address', async () => {
+      expect(await entrypoint.getVault()).to.be.eq(vault.address);
+    });
+
+    it('uses the authorizer of the vault', async () => {
+      expect(await entrypoint.getAuthorizer()).to.equal(authorizer.address);
+    });
+
+    it('tracks authorizer changes in the vault', async () => {
+      const action = await actionId(vault, 'setAuthorizer');
+      await authorizer.connect(admin).grantPermissions([action], admin.address, [ANY_ADDRESS]);
+
+      await vault.connect(admin).setAuthorizer(other.address);
+
+      expect(await entrypoint.getAuthorizer()).to.equal(other.address);
+    });
+  });
+
+  describe('performAction', () => {
+    let action: string;
+    let target: string;
+    let calldata: string;
+    let expectedResult: string;
+
+    sharedBeforeEach('prepare action', async () => {
+      action = await actionId(adaptor, 'getProtocolFeesCollector', vault.interface);
+
+      target = vault.address;
+      calldata = vault.interface.encodeFunctionData('getProtocolFeesCollector');
+
+      expectedResult = defaultAbiCoder.encode(['address'], [await vault.getProtocolFeesCollector()]);
+    });
+
+    context('when caller is authorized globally', () => {
+      sharedBeforeEach('authorize caller globally', async () => {
+        await authorizer.connect(admin).grantPermissions([action], grantee.address, [ANY_ADDRESS]);
+      });
+
+      it('performs the expected function call', async () => {
+        const value = await entrypoint.connect(grantee).callStatic.performAction(target, calldata);
+        expect(value).to.be.eq(expectedResult);
+      });
+    });
+
+    context('when caller is authorized locally on target', () => {
+      sharedBeforeEach('authorize caller on target locally', async () => {
+        await authorizer.connect(admin).grantPermissions([action], grantee.address, [vault.address]);
+      });
+
+      it('performs the expected function call', async () => {
+        const value = await entrypoint.connect(grantee).callStatic.performAction(target, calldata);
+
+        expect(value).to.be.eq(expectedResult);
+      });
+    });
+
+    context('when caller is authorized locally on a different target', () => {
+      sharedBeforeEach('authorize caller on different target locally', async () => {
+        await authorizer.connect(admin).grantPermissions([action], grantee.address, [other.address]);
+      });
+
+      it('reverts', async () => {
+        await expect(entrypoint.connect(grantee).performAction(target, calldata)).to.be.revertedWith(
+          'SENDER_NOT_ALLOWED'
+        );
+      });
+    });
+
+    context('when caller is not authorized', () => {
+      it('reverts', async () => {
+        await expect(entrypoint.connect(other).performAction(target, calldata)).to.be.revertedWith(
+          'SENDER_NOT_ALLOWED'
+        );
+      });
+    });
+
+    context('when calldata is invalid', () => {
```

Comment:
I have one in the dependent PR

Reply:
Should we remove these from this test? I don't have a strong opinion; it's a bit redundant but I don't think it's bad either.

Reply:
We've gotten in trouble before removing "redundant" tests (e.g., having ComposableStablePool not test `setSwapFeePercentage` because BasePool tested it). I don't think it hurts anything.

We should add a comment saying why though - it rejects them because you have to go through the entrypoint now.